components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -394,8 +394,18 @@ public String buildIDToken(OAuthAuthzReqMessageContext request, OAuth2AuthorizeR
         jwtClaimsSet.setClaim("azp", request.getAuthorizationReqDTO().getConsumerKey());
         jwtClaimsSet.setExpirationTime(new Date(curTimeInMillis + lifetimeInMillis));
         jwtClaimsSet.setIssueTime(new Date(curTimeInMillis));
-        if (request.getAuthorizationReqDTO().getAuthTime() != 0) {
-            jwtClaimsSet.setClaim("auth_time", request.getAuthorizationReqDTO().getAuthTime() / 1000);
+        long authTime = 0;
+        if (StringUtils.isNotEmpty(tokenRespDTO.getAccessToken())) {
```

Comment:
Are there any other essential claims we need to handle?
If so better to have a separate method for all of them to avoid duplication.